src/nft/components/bag/BagFooter.tsx

```diff
@@ -179,34 +343,62 @@ export const BagFooter = ({ totalEthPrice, bagStatus, fetchAssets, eventProperti
       }
       disabled = false
       buttonText = <Trans>Connect wallet</Trans>
+    } else if (usingPayWithAnyToken && tradeState !== TradeState.VALID) {
+      disabled = true
+      buttonText = <Trans>Fetching Route</Trans>
```

Comment:
While fetching route I can still enter the purchasing state and fire off multiple txs
![multipleTxs](https://user-images.githubusercontent.com/11512321/217347439-99a1925e-560c-4d11-abb3-2b335aa144d7.gif)

Reply:
this is actually a bug on main .... fixed